packages/grafana-runtime/src/services/RuntimeDataSource.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  DataQuery,
+  DataSourceApi,
+  DataSourceInstanceSettings,
+  PluginType,
+  TestDataSourceResponse,
+} from '@grafana/data';
+
+export abstract class RuntimeDataSource<TQuery extends DataQuery = DataQuery> extends DataSourceApi<TQuery> {
+  public instanceSettings: DataSourceInstanceSettings;
+
+  public constructor(pluginId: string, uid: string) {
+    const instanceSettings: DataSourceInstanceSettings = {
+      name: 'RuntimeDataSource-' + pluginId,
+      uid: uid,
+      type: pluginId,
+      id: 1,
```

Comment:
Should this ID also auto-increment/ensure some kind of uniqueness? Or is it not used?

Reply:
not used anywhere